feat: Migrate utilities from UrlContextProvider to Zustand stores (resolves #269; fixes #294; resolves #306).

zzxthehappiest opened the pull request:
<!-- markdownlint-disable MD012 -->

<!--
Set the PR title to a meaningful commit message that:

* is in imperative form.
* follows the Conventional Commits specification (https://www.conventionalcommits.org).
  * See https://github.com/commitizen/conventional-commit-types/blob/master/index.json for possible
    types.

Example:

fix: Don't add implicit wildcards ('*') at the beginning and the end of a query (fixes #390).
-->

# Description

<!-- Describe what this request will change/fix and provide any details necessary for reviewers. -->

#269 

# Checklist

<!-- Ensure each item below is satisfied and indicate so by inserting an `x` within each `[ ]`. -->

* [x] The PR satisfies the [contribution guidelines][yscope-contrib-guidelines].
* [x] This is a breaking change and that has been indicated in the PR title, OR this isn't a
  breaking change.
* [x] Necessary docs have been updated, OR no docs need to be updated.

# Validation performed

<!-- Describe what tests and validation you performed on the change. -->

1. All basic operations related to URL parameters (`isPrettified`, `logEventNum`, query-related, `filePath`).
2. Remove the hash parameters to see if they are reset to the default value (except the `logEventNum` because its default value is 0, which is not a valid one).

[yscope-contrib-guidelines]: https://docs.yscope.com/dev-guide/contrib-guides-overview.html


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Summary by CodeRabbit

- **Refactor**
  - Streamlined URL parameter management by removing React context and centralizing logic in dedicated utility functions and Zustand stores.
  - Deleted legacy context provider and related context store, simplifying state management.
  - Unified handling of log event numbers and prettify state within a single store for improved synchronization.
  - Refactored component event handlers and callbacks to use memoization and direct store state access for better performance and clarity.

- **Bug Fixes**
  - Improved synchronization between URL parameters and application state, ensuring accurate updates when navigating or sharing links.

- **New Features**
  - Added robust utilities for copying permalinks and updating URL parameters without reloading the page.

- **Style**
  - Enhanced code consistency and maintainability by simplifying internal component logic and imports.

- **Documentation**
  - Updated types to reflect non-nullable log event numbers and improved typings for URL parameter utilities.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

junhaoliao commented on src/stores/viewStore.ts:
```diff
@@ -43,7 +43,7 @@ interface ViewStoreActions {
 
     loadPageByAction: (navAction: NavigationAction) => void;
-    updateIsPrettified: (newIsPrettified: boolean) => void;
+    setIsPrettified: (newIsPrettified: boolean) => void;
```
(i understand this is a WIP PR and the changes are not finalized yet)

for all Zustand store actions that do not simply change the value of a state, we prefer naming them with a `updatePrefix`. @coderabbitai create an issue for adding this in the dev docs

zzxthehappiest commented on src/utils/url.ts:
```diff
@@ -1,3 +1,135 @@
+    const {filePath: newFilePath} = updates;
+
+    for (const [key, value] of Object.entries(updates)) {
+        if (SEARCH_PARAM_NAMES.FILE_PATH as string === key) {
```
If remove this, it will have this lint error:
```
 error  The two values in this comparison do not have a shared enum type  @typescript-eslint/no-unsafe-enum-comparison
```

zzxthehappiest commented:
I think if we add unit tests for `url.ts` then this PR will exceed 400 lines limitation.

zzxthehappiest commented:
> Thanks for quickly putting up the changes.
> 
> Could you also help integrating the changes in [junhaoliao@b972113](https://github.com/junhaoliao/yscope-log-viewer/commit/b972113bf7115b80586c552f86fe84f1733527a7) into the current PR? Those will fix #294
> 
> (my fix is to align the existing parameters parsing logic. in my changes, we might also need to make sure the parsed values are of `string` type before calling `new URLSearchParams()`. That said, maybe the code should be completely rewritten - the for-loops can go through the `currentParams` objects to insert values into an empty `new URLSearchParams()`. if you think there's a better way to fix #294, feel free to make your own changes

I modified the `filterNullValues` to `filterNullValuesToStrings` because the types of both keys and values need to be `string` (otherwise Webstorm told me it was an error), so I converted all values to strings.

Henry8192 commented on src/components/AppController.tsx:
```diff
@@ -78,40 +78,73 @@ interface AppControllerProps {
+        return () => {
+            window.removeEventListener("hashchange", handleHashChange);
+        };
+    }, [setFileSrc,
```
```suggestion
    }, [
        setFileSrc,
```

Henry8192 commented on src/components/AppController.tsx:
```diff
@@ -78,40 +78,73 @@ interface AppControllerProps {
+    }, [setFileSrc,
         setLogEventNum,
-    ]);
+        updateIsPrettified]);
```
```suggestion
        updateIsPrettified
    ]);
```

Henry8192 commented on src/components/AppController.tsx:
```diff
@@ -78,40 +78,73 @@ interface AppControllerProps {
     // Synchronize `isPrettifiedRef` with `isPrettified`.
     useEffect(() => {
-        isPrettifiedRef.current = isPrettified ?? false;
-        setIsPrettified(isPrettifiedRef.current);
+        isPrettifiedRef.current = isPrettified;
     }, [
         isPrettified,
-        setIsPrettified,
+    ]);
```
```suggestion
```

junhaoliao replied:
see the comment at https://github.com/y-scope/yscope-log-viewer/pull/286/files#r2108340456 , i believe this useEffect hook can be removed

Henry8192 commented on src/components/AppController.tsx:
```diff
@@ -78,40 +78,73 @@ interface AppControllerProps {
-        setIsPrettified,
+    ]);
+
+    // Synchronize `logEventNumRef` with `logEventNum`.
+    useEffect(() => {
+        logEventNumRef.current = logEventNum;
+    }, [
+        logEventNum,
     ]);
```
```suggestion
    }, [logEventNum]);
```

junhaoliao replied:
see the comment at https://github.com/y-scope/yscope-log-viewer/pull/286/files#r2108340456 , i believe this useEffect hook can be removed

Henry8192 commented on src/contexts/UrlContextProvider.tsx:
We should delete this file instead of clearing its content

Henry8192 commented on src/components/CentralContainer/Sidebar/SidebarTabs/SearchTabPanel/Result.tsx:
```diff
@@ -36,8 +37,10 @@ const Result = ({logEventNum, message, matchRange}: ResultProps) => {
         message.slice(matchRange[1]),
     ];
+    const setLogEventNum = useViewStore((state) => state.setLogEventNum);
     const handleResultButtonClick = () => {
         updateWindowUrlHashParams({logEventNum});
+        setLogEventNum(logEventNum);
     };
```
```suggestion
    const handleResultButtonClick = useCallback(() => {
        updateWindowUrlHashParams({logEventNum});
        setLogEventNum(logEventNum);
    }, [logEventNum]);
```
This is my fault. We should use `useCallback()` hook for this function to avoid repetitive function rendering. I'm not adding `setLogEventNum` to the dependency array, since my previous comment suggests we should use `getState()` to get `setLogEventNum` inside this function.

Henry8192 commented on src/components/Editor/index.tsx:
```diff
@@ -138,16 +135,22 @@ const Editor = () => {
     const logData = useViewStore((state) => state.logData);
+
+    const logEventNum = useViewStore((state) => state.logEventNum);
+    const setLogEventNum = useViewStore((state) => state.setLogEventNum);
```
use `getState()`

junhaoliao replied:
(for clarity) we want to retrieve those actions, whose definitions never change directly inside useEffect hooks. see https://github.com/y-scope/yscope-log-viewer/pull/286/files#r2108340456 for an example

Henry8192 commented on src/components/Editor/index.tsx:
```diff
@@ -176,18 +179,22 @@ const Editor = () => {
                 break;
         }
-    }, [loadPageByAction]);
+    }, [loadPageByAction,
+        updateIsPrettified]);
```
```suggestion
    }, [
        loadPageByAction,
        updateIsPrettified
    ]);
```

Henry8192 commented on src/components/StatusBar/index.tsx:
```diff
@@ -42,16 +40,19 @@ const handleCopyLinkButtonClick = () => {
-    const {isPrettified, logEventNum} = useContext(UrlContext);
+    const isPrettified = useViewStore((state) => state.isPrettified);
+    const logEventNum = useViewStore((state) => state.logEventNum);
+    const setIsPrettified = useViewStore((state) => state.updateIsPrettified);
```
use `getState()`

Henry8192 replied:
variable should be named to `updateIsPrettified`. I had a discussion with @junhaoliao, we are doing more than simply setting `isPrettified` in this function, so it's better to rename this to `updateIsPrettified`.

Henry8192 commented on src/stores/viewStore.ts:
```diff
@@ -225,9 +229,12 @@ const useViewStore = create<ViewState>((set, get) => ({
+            logEventNum: newLogEventNum,
         });
+        const {setLogEventNum} = get();
+        setLogEventNum(newLogEventNum ?? 0);
```
Let's change the type of PageData and change `logEventNum: Nullable<number>` to `logEventNum: number`. Then assign `EMPTY_PAGE_RESP.logEventNum = 0`. To me, it is not necessary to use nullable here. @junhaoliao

junhaoliao replied:
sure. let's give that a try, and do [sanity checks](https://docs.yscope.com/yscope-log-viewer/main/dev-guide/contributing-validation.html#manual-testing) after to make sure things don't break

junhaoliao commented on src/components/AppController.tsx:
```diff
@@ -78,40 +78,73 @@ interface AppControllerProps {
 
     // Refs
-    const isPrettifiedRef = useRef<boolean>(isPrettified ?? false);
+    const isPrettifiedRef = useRef<boolean>(isPrettified);
     const logEventNumRef = useRef(logEventNum);
```
fwiw, those refs are no longer needed. thanks to Zustand, in the useEffect hooks we can read the latest value of some Zustand state variable using
```
const {isPrettified} = useViewStore.getState();
```
for example

junhaoliao commented on src/components/AppController.tsx:
```diff
@@ -78,40 +78,73 @@ interface AppControllerProps {
+                updateIsPrettified(hashParams.isPrettified);
+            }
+
+            // Also check search params to handle initial page load and maintain full URL state
+            const searchParams = getWindowUrlSearchParams();
+
+            if (null !== searchParams.filePath) {
+                setFileSrc(searchParams.filePath);
+            }
```
also if we apply the Zustand state retrieval convention as suggested by https://github.com/y-scope/yscope-log-viewer/pull/286/files#r2107557054 , i believe we can declare `handleHashChange` outside of (before) `AppController()`

junhaoliao commented on src/components/AppController.tsx:
```diff
@@ -124,8 +157,15 @@ const AppController = ({children}: AppControllerProps) => {
 
-        if (updateUrlIfEventOnPage(clampedLogEventNum, logEventNumsOnPage)) {
+        const {
+            isUpdated,
+            nearestLogEventNum,
```
this is minor but i prefer the destructured variables on their own lines (as currently written)

junhaoliao commented on src/stores/logFileStore.ts:
```diff
@@ -38,6 +39,7 @@ interface LogFileValues {
 
 interface LogFileActions {
     setFileName: (newFileName: string) => void;
+    setFileSrc: (newFileSrc: Nullable<FileSrcType>) => void;
```
we may not need action `setFileSrc`

junhaoliao commented on src/components/CentralContainer/Sidebar/SidebarTabs/SearchTabPanel/Result.tsx:
```diff
@@ -36,8 +37,10 @@ const Result = ({logEventNum, message, matchRange}: ResultProps) => {
         message.slice(...matchRange),
         message.slice(matchRange[1]),
     ];
+    const setLogEventNum = useViewStore((state) => state.setLogEventNum);
```
(for clarity) we want to retrieve those actions, whose definitions never change directly inside useEffect hooks. see https://github.com/y-scope/yscope-log-viewer/pull/286/files#r2108340456 for an example

junhaoliao commented on src/components/Editor/index.tsx:
```diff
@@ -138,16 +135,22 @@ const Editor = () => {
     );
     const editorRef = useRef<Nullable<monaco.editor.IStandaloneCodeEditor>>(null);
-    const isPrettifiedRef = useRef<boolean>(isPrettified ?? false);
+    const isPrettifiedRef = useRef<boolean>(isPrettified);
```
similar to https://github.com/y-scope/yscope-log-viewer/pull/286/files#r2108340456 , we may not no long need to copy states into refs. instead, we can directly read them from the Zustand stores.

junhaoliao commented on src/components/Editor/index.tsx:
```diff
@@ -138,16 +135,22 @@ const Editor = () => {
-    const {isPrettified, logEventNum} = useContext(UrlContext);
 
     const [lineNum, setLineNum] = useState<number>(1);
     const beginLineNumToLogEventNumRef = useRef<BeginLineNumToLogEventNumMap>(
```
similar to https://github.com/y-scope/yscope-log-viewer/pull/286/files#r2108340456 , we may not no long need to copy states into refs. instead, we can directly read them from the Zustand stores.

junhaoliao commented on src/components/Editor/index.tsx:
```diff
@@ -255,7 +262,8 @@ const Editor = () => {
+        setLogEventNum(newLogEventNum);
+    }, [setLogEventNum]);
 
     // Synchronize `beginLineNumToLogEventNumRef` with `beginLineNumToLogEventNum`.
```
similar to https://github.com/y-scope/yscope-log-viewer/pull/286/files#r2108340456 , we may not no long need to copy states into refs. instead, we can directly read them from the Zustand stores.

junhaoliao commented on src/components/Editor/index.tsx:
```diff
@@ -264,7 +272,7 @@ const Editor = () => {
 
     // Synchronize `isPrettifiedRef` with `isPrettified`.
```
similar to https://github.com/y-scope/yscope-log-viewer/pull/286/files#r2108340456 , we may not no long need to copy states into refs. instead, we can directly read them from the Zustand stores.

junhaoliao commented on src/components/AppController.tsx:
```diff
@@ -144,13 +184,14 @@ const AppController = ({children}: AppControllerProps) => {
     ]);
 
-    // On `filePath` update, load file.
+    // On `fileSrc` update, load file.
```
I believe this useEffect hook can be removed completely

junhaoliao replied:
see https://github.com/y-scope/yscope-log-viewer/pull/286/files#r2108403681 for the rationale

anlowee commented:
I think we are a bit late for the original plan. Let's speed it up.

junhaoliao commented on src/components/AppController.tsx:
```diff
@@ -79,43 +92,47 @@ interface AppControllerProps {
     const beginLineNumToLogEventNum = useViewStore((state) => state.beginLineNumToLogEventNum);
-    const setIsPrettified = useViewStore((state) => state.updateIsPrettified);
+
+    const {loadFile} = useLogFileStore.getState();
```
can we also move this into the useEffect hook since its definition won't change?

zzxthehappiest replied:
Fixed!

junhaoliao commented on src/components/AppController.tsx:
```diff
@@ -79,43 +92,47 @@ interface AppControllerProps {
+    const numEvents = useLogFileStore((state) => state.numEvents);
+
+    const {logEventNum} = useViewStore.getState();
+    const {setLogEventNum} = useViewStore.getState();
```
ditto - can this be moved into the hook so we don't have to add it to the dependency array?

zzxthehappiest replied:
Fixed!

junhaoliao commented:
no rush to get to the suggestions yet as I'm not done with the review. i'll post the complete review tonight China time

zzxthehappiest commented on src/components/Editor/index.tsx:
```diff
@@ -138,16 +135,21 @@ const Editor = () => {
     const {mode, systemMode} = useColorScheme();
 
     const beginLineNumToLogEventNum = useViewStore((state) => state.beginLineNumToLogEventNum);
+    const {isPrettified} = useViewStore.getState();
+    const updateIsPrettified = useViewStore((state) => state.updateIsPrettified);
```
For the first issue, the `updateIsPrettifed` is the setter for `isPrettifed` but it is doing extra things than just setting the value of `isPrettified`. And for `setLogEventNum` there is no need to subscribe it because it will never change, so I use `getState()`.

For the second issue, all other state variables are declared at the beginning, so I don't think it's reasonable to apply your suggestions.

zzxthehappiest commented on src/components/CentralContainer/Sidebar/SidebarTabs/SearchTabPanel/QueryInputBox.tsx:
```diff
@@ -33,17 +34,23 @@ const QueryInputBox = () => {
 
     const handleQueryInputChange = (ev: React.ChangeEvent<HTMLTextAreaElement>) => {
-        setQueryString(ev.target.value);
+        const newQueryString = ev.target.value;
+        updateWindowUrlHashParams({queryString: newQueryString});
+        setQueryString(newQueryString);
         startQuery();
```
There are no errors thrown, so no need to do this.

junhaoliao commented on src/components/Editor/index.tsx:
```diff
@@ -1,8 +1,8 @@
 /* eslint max-lines: ["error", 350] */
-/* eslint max-lines-per-function: ["error", 170] */
+/* eslint max-lines-per-function: ["error", 200] */
+/* eslint max-statements: ["error", 30] */
```
if you agree with the below suggestions, i believe we can remove those rule overrides
```suggestion
```

junhaoliao commented on src/components/Editor/index.tsx:
```diff
@@ -138,16 +135,21 @@ const Editor = () => {
-    const isPrettifiedRef = useRef<boolean>(isPrettified ?? false);
     const isMouseDownRef = useRef<boolean>(false);
     const pageSizeRef = useRef(getConfig(CONFIG_KEY.PAGE_SIZE));
 
```
can we pull `handleEditorCustomAction` into the global scope since its definition does not need to be changed depending on the Zustand states?
```
/**
 * Handles custom actions in the editor based on the action name.
 *
 * @param editor
 * @param actionName
 */
const handleEditorCustomAction = (
    editor: monaco.editor.IStandaloneCodeEditor,
    actionName: ACTION_NAME
) => {
    switch (actionName) {
        case ACTION_NAME.FIRST_PAGE:
        case ACTION_NAME.PREV_PAGE:
        case ACTION_NAME.NEXT_PAGE:
        case ACTION_NAME.LAST_PAGE: {
            const {loadPageByAction} = useViewStore.getState();
            loadPageByAction({code: actionName, args: null});
            break;
        }
        case ACTION_NAME.PAGE_TOP:
            goToPositionAndCenter(editor, {lineNumber: 1, column: 1});
            break;
        case ACTION_NAME.PAGE_BOTTOM: {
            const lineCount = editor.getModel()?.getLineCount();
            if ("undefined" === typeof lineCount) {
                break;
            }
            goToPositionAndCenter(editor, {lineNumber: lineCount, column: 1});
            break;
        }
        case ACTION_NAME.COPY_LOG_EVENT: {
            const {beginLineNumToLogEventNum} = useViewStore.getState();
            handleCopyLogEventAction(editor, beginLineNumToLogEventNum);
            break;
        }
        case ACTION_NAME.TOGGLE_PRETTIFY: {
            const {isPrettified, updateIsPrettified} = useViewStore.getState();
            const newIsPrettified = !isPrettified;
            updateWindowUrlHashParams({
                [HASH_PARAM_NAMES.IS_PRETTIFIED]: newIsPrettified,
            });
            updateIsPrettified(newIsPrettified);
            break;
        }
        case ACTION_NAME.TOGGLE_WORD_WRAP:
            handleToggleWordWrapAction(editor);
            break;
        default:
            break;
    }
};
```

junhaoliao commented on src/components/Editor/index.tsx:
```diff
@@ -138,16 +135,21 @@ const Editor = () => {
     const logData = useViewStore((state) => state.logData);
+
+    const logEventNum = useViewStore((state) => state.logEventNum);
+    const {setLogEventNum} = useViewStore.getState();
```
```suggestion
```
see the suggestion about retrieving the action directly inside the useCallback wrapped function

junhaoliao commented on src/components/Editor/index.tsx:
```diff
@@ -255,18 +264,14 @@ const Editor = () => {
         }
         updateWindowUrlHashParams({logEventNum: newLogEventNum});
-    }, []);
+        setLogEventNum(newLogEventNum);
+    }, [setLogEventNum]);
```
```suggestion
        const {setLogEventNum} = useViewStore.getState();
        setLogEventNum(newLogEventNum);
    }, []);
```

junhaoliao commented on src/components/CentralContainer/Sidebar/SidebarTabs/SearchTabPanel/Result.tsx:
```diff
@@ -1,9 +1,12 @@
 } from "@mui/joy";
 
-import {updateWindowUrlHashParams} from "../../../../../contexts/UrlContextProvider";
+import useViewStore from "../../../../../stores/viewStore.ts";
```
```suggestion
import useViewStore from "../../../../../stores/viewStore";
```

junhaoliao commented on src/components/CentralContainer/Sidebar/SidebarTabs/SearchTabPanel/Result.tsx:
```diff
@@ -1,9 +1,12 @@
 
-import {updateWindowUrlHashParams} from "../../../../../contexts/UrlContextProvider";
+import useViewStore from "../../../../../stores/viewStore.ts";
+import {updateWindowUrlHashParams} from "../../../../../utils/url.ts";
```
```suggestion
import {updateWindowUrlHashParams} from "../../../../../utils/url";
```

junhaoliao commented on src/components/CentralContainer/Sidebar/SidebarTabs/SearchTabPanel/index.tsx:
```diff
@@ -12,12 +12,12 @@ import ShareIcon from "@mui/icons-material/Share";
     TAB_DISPLAY_NAMES,
     TAB_NAME,
 } from "../../../../../typings/tab";
+import {copyPermalinkToClipboard} from "../../../../../utils/url.ts";
```
i agree with the rabbit

junhaoliao commented on src/components/StatusBar/index.tsx:
```diff
@@ -10,17 +8,17 @@ import {
-} from "../../contexts/UrlContextProvider";
 import useLogFileStore from "../../stores/logFileStore";
 import useUiStore from "../../stores/uiStore";
+import useViewStore from "../../stores/viewStore.ts";
```
```suggestion
import useViewStore from "../../stores/viewStore";
```

junhaoliao commented on src/components/StatusBar/index.tsx:
```diff
@@ -10,17 +8,17 @@ import {
 import {HASH_PARAM_NAMES} from "../../typings/url";
 import {ACTION_NAME} from "../../utils/actions";
 import {isDisabled} from "../../utils/states";
+import {
+    copyPermalinkToClipboard,
+    updateWindowUrlHashParams,
+} from "../../utils/url.ts";
```
```suggestion
import {
    copyPermalinkToClipboard,
    updateWindowUrlHashParams,
} from "../../utils/url";
```

junhaoliao commented on src/stores/viewStore.ts:
```diff
@@ -16,7 +15,7 @@ import {
 } from "../utils/actions";
 import {clamp} from "../utils/math";
-import useContextStore, {CONTEXT_STORE_DEFAULT} from "./contextStore";
+import {updateWindowUrlHashParams} from "../utils/url.ts";
```
```suggestion
import {updateWindowUrlHashParams} from "../utils/url";
```

junhaoliao commented on src/utils/js.ts:
```diff
@@ -1,9 +1,25 @@
+import {Nullable} from "../typings/common.ts";
```
```suggestion
import {Nullable} from "../typings/common";
```

junhaoliao commented on src/utils/url.ts:
```diff
@@ -1,3 +1,40 @@
+/* eslint max-lines: ["error", 350] */
+import {NullableProperties} from "../typings/common.ts";
```
```suggestion
import {NullableProperties} from "../typings/common";
```

junhaoliao commented on src/utils/url.ts:
```diff
@@ -1,3 +1,40 @@
+/* eslint max-lines: ["error", 350] */
+import {NullableProperties} from "../typings/common.ts";
+import {
+    HASH_PARAM_NAMES,
+    SEARCH_PARAM_NAMES,
+    UrlHashParams,
+    UrlHashParamUpdatesType,
+    UrlSearchParams,
+    UrlSearchParamUpdatesType,
+} from "../typings/url.ts";
```
```suggestion
import {
    HASH_PARAM_NAMES,
    SEARCH_PARAM_NAMES,
    UrlHashParams,
    UrlHashParamUpdatesType,
    UrlSearchParams,
    UrlSearchParamUpdatesType,
} from "../typings/url";
```

junhaoliao reviewed (changes requested):
we don't have to specify the file extension in the imports

zzxthehappiest commented:
@junhaoliao Sorry for the delay. I am on a business since last week and get home late recently. I fixed most of comments except [this one](https://github.com/y-scope/yscope-log-viewer/pull/286#discussion_r2120634361). I found a behaviour that when clicking the "Prettify" button, there will be a `isPrettified=true`. When I delete the  `isPrettified=true` the logs remain prettified because the `isPrettified` is null so its state won't be updated. I wonder do we want this or do we want when a parameter is missing we set it with the default value.

I will definitely finish all things by tomorrow. I need to take a look of the changed code again to ensure all of them apply the comments.

zzxthehappiest commented on src/utils/url.ts:
For this one, even I put the the new line to group them, the lint will still alphabetize them.

junhaoliao commented on src/stores/viewStore.ts:
```diff
@@ -109,13 +111,9 @@ const useViewStore = create<ViewState>((set, get) => ({
         const {setUiState} = useUiStore.getState();
         setUiState(UI_STATE.FAST_LOADING);
 
-        const {clearQuery} = useQueryStore.getState();
-        clearQuery();
```
somehow this comment was missed in the last batch -

good catch! let's also remove the one in logFileStore.ts: loadFile(); otherwise, this link won't load the initial query: http://localhost:3010/?filePath=/test/test.clp.zst#logEventNum=999&queryString=hi

junhaoliao commented:
@zzxthehappiest Again thanks for your contribution! I'm merging this now
